cirq-google/cirq_google/workflow/quantum_runtime.py

```diff
@@ -180,26 +212,38 @@ def execute(
         # coverage: ignore
         raise ValueError("Please provide a non-empty `base_data_dir`.")
 
-    shared_rt_info = SharedRuntimeInfo(run_id=run_id)
+    sampler = rt_config.processor.get_sampler()
+    device = rt_config.processor.get_device()
+
+    shared_rt_info = SharedRuntimeInfo(
+        run_id=run_id,
+        device=device,
+    )
     executable_results = []
 
     saver = _FilesystemSaver(base_data_dir=base_data_dir, run_id=run_id)
     saver.initialize(rt_config, shared_rt_info)
 
-    sampler = rt_config.processor.get_sampler()
     logger = _PrintLogger(n_total=len(executable_group))
     logger.initialize()
+
+    rs = np.random.RandomState(rt_config.random_seed)
```

Comment:
Question: Should we give the seed to the `rt_config` or the randomstate generator itself ? It seems like down the line if we kept creating `randomState`s off of this seed we'd get a lot of duplicate random numbers.

Reply:
two part answer

1. `rt_config` needs to be serializable and put in a database so we can groupby and agg the results. Seed is (much!) easier than the internal state of a random state
2. We don't create >1 randomstate based on this seed. We create one at the top of the function (here!) and it gets passed to any subroutines that require randomness.